core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -627,7 +627,17 @@ object Chunk extends CollectorK[Chunk] with ChunkCompanionPlatform {
 
   /** Creates a chunk backed by a `Chain`. */
   def chain[O](c: Chain[O]): Chunk[O] =
-    seq(c.toList)
+    if (c.isEmpty) empty
```

Comment:
this is the generic `Iterable` implementation and seems strictly better than converting to `List` first, which is what we were doing before.